server/src/main/java/org/apache/uniffle/server/ShuffleDataFlushEvent.java

```diff
@@ -105,6 +109,22 @@ public void increaseRetryTimes() {
     retryTimes.incrementAndGet();
   }
 
+  public boolean isPended() {
+    return isPended;
+  }
+
+  public void markPended() {
+    isPended = true;
+  }
+
+  public Storage getUnderStorage() {
+    return underStorage;
```

Comment:
Why do we need `underStorage`?

Reply:
For the getting concrete storage when updating metrics. In original logic, the storage will be gotten by invoking selectStorage method. This is unreasonable. 

We should guarantee that StorageManager#selectStorage only could be invoked before writing. After written, the storage has been bound to event, we shouldn’t invoke selectStorage again.

Reply:
OK.

Reply:
If you could fetch this PR into your local IDEA, you could find the invoking chain. This will help understand. Although this PR scope  may be a little bit scattered, to fix some unreasonable design.